cuda/solver/gmres_kernels.cu

```diff
@@ -720,15 +714,6 @@ void solve_x(std::shared_ptr<const CudaExecutor> exec,
         y->get_stride(), as_cuda_type(before_preconditioner->get_values()),
         stride_before_preconditioner,
         as_cuda_type(final_iter_nums->get_const_data()));
-
-    preconditioner->apply(before_preconditioner.get(),
-                          after_preconditioner.get());
-
-    auto one_op =
-        initialize<matrix::Dense<ValueType>>({one<ValueType>()}, exec);
-    x->add_scaled(one_op.get(), after_preconditioner.get());
-    // Solve x
-    // x = x + preconditioner_ * krylov_bases * y
```

Comment:
I think the `Solve x` comment can stay (we might have to replace it with `Solve for before_preconditioner` or similar).

Reply:
Welle there is no more `preconditioner` here, so it's not really the same operation anymore, we only have krylov_bases * y I believe?

Also, I will change the name of this function to `calculate_qy` to beging with, as it is no more `solve_x`.